src/sources/util/http/prelude.rs

```diff
@@ -261,21 +265,43 @@ async fn handle_request(
     events: Result<Vec<Event>, ErrorMessage>,
     acknowledgements: bool,
     response_code: StatusCode,
+    response_body_key: OptionalTargetPath,
     mut out: SourceSender,
 ) -> Result<impl warp::Reply, Rejection> {
     match events {
         Ok(mut events) => {
-            let receiver = BatchNotifier::maybe_apply_to(acknowledgements, &mut events);
+            let mut response = response_code.into_response();
+
+            if response_body_key.path.is_some() {
+                let cloned = events.clone();
```

Comment:
is there a better way than cloning the events? I'm worried this might have performance implications